File: 알고리즘/전승민/TileOrnament/src/TileOrnament.java
public class TileOrnament {
    public long solution(int N) {
        long[] dp = new long[81];
        dp[0] = dp[1] = 1;
        for(int i=2; i<=N; i++){
            dp[i] = dp[i-1] + dp[i-2];
        }
        return dp[N-1]*2+dp[N]*2;
    }
}

// https://programmers.co.kr/learn/courses/30/lessons/43104

File: 알고리즘/박기태/tileOrnament/tileOrnament.js
// 초기에 작성한 코드 최적화
function solution(N) {
  const arr = [1, 1];

  for (let i = 0; i < N; i++) {
    arr.push(arr[0] + arr[1]);
    arr.shift();
  }
  return arr.pop() * 2; // arr[N+1] * 2
}

// Recommended & 20%~ Fast
function solution(N) {
  const arr = [1, 1];

  for (let i = 0; i < N; i++) arr.push(arr[i] + arr[i + 1]);

  return arr.pop() * 2; // arr[N+1] * 2
}

// https://programmers.co.kr/learn/courses/30/lessons/43104
